# **Chinook Database Analysis Assignment**

## **Scenario:**

You are a data analyst at **MusicWave Inc.**, a company that specializes in music distribution and sales. Your company uses the Chinook database to manage its music collection, sales, and customer information. Recently, your manager has tasked you with analyzing the database to extract valuable insights that can help the company make informed business decisions.

## **Introduction to the Chinook Database:**

The Chinook database is a sample database that represents a digital media store. It includes tables for:
- **Artists**: Information about the music artists.
- **Albums**: Information about the albums, each linked to an artist.
- **Tracks**: Information about individual tracks, including which album they belong to.
- **Customers**: Information about the customers.
- **Invoices**: Information about sales transactions.
- **InvoiceLines**: Detailed information about each item in an invoice.
- **Genres**: Information about the genres of the tracks.
- **MediaTypes**: Information about the types of media.
- **Employees**: Information about the employees.

The database schema allows for complex queries to analyze various aspects of the music store's operations.

## **Getting Started:**

### **Download the Chinook Database**

1. Download the Chinook SQLite database from [here](https://github.com/lerocha/chinook-database/raw/master/ChinookDatabase/DataSources/Chinook_Sqlite.sqlite).

2. Upload the `Chinook_Sqlite.sqlite` file to your Google Colab environment.

### **Setting Up the Environment**

Run the following cells to set up your environment and connect to the Chinook database.

```python
# Install sqlite3 library (if not already installed)
!pip install sqlite3

# Import necessary libraries
import sqlite3
import pandas as pd

# Connect to the SQLite database
conn = sqlite3.connect('Chinook_Sqlite.sqlite')


In [163]:
import sqlite3
import pandas as pd

# Connect to the SQLite database
conn = sqlite3.connect('Chinook_Sqlite.sqlite')

# Execute a query
query = "SELECT * FROM sqlite_master where type='table'"

df = pd.read_sql_query(query, conn)
df[['type','name']]



type           name
0   table          Album
1   table         Artist
2   table       Customer
3   table       Employee
4   table          Genre
5   table        Invoice
6   table    InvoiceLine
7   table      MediaType
8   table       Playlist
9   table  PlaylistTrack
10  table          Track

In [164]:
# Read the first five rows from the Album table
df = pd.read_sql_query("SELECT * FROM album limit 5", conn)
df

AlbumId                                  Title  ArtistId
0        1  For Those About To Rock We Salute You         1
1        2                      Balls to the Wall         2
2        3                      Restless and Wild         2
3        4                      Let There Be Rock         1
4        5                               Big Ones         3

### **Question 1: List All Tables**

**Scenario:**  
Your manager, Alex, is new to the team and is trying to understand the structure of the Chinook database. Alex asks you to provide a list of all the tables in the database to get an overview of the available data.

**Task:**  
Write a query to list all tables in the Chinook database.

In [165]:
query = "SELECT * FROM sqlite_master where type='table'"

df = pd.read_sql_query(query, conn)
df[['name']]

name
0           Album
1          Artist
2        Customer
3        Employee
4           Genre
5         Invoice
6     InvoiceLine
7       MediaType
8        Playlist
9   PlaylistTrack
10          Track

### **Question 2: Retrieve Top 10 Records from the Albums Table**

**Scenario:**  
Alex wants to familiarize himself with the kind of albums stored in the database. He asks you to fetch a sample of the album records to get a better idea of the data.

**Task:**  
Write a query to retrieve the top 10 records from the Albums table.

---


In [166]:
query = 'SELECT * FROM Album LIMIT 10'

df = pd.read_sql_query(query,conn)
df

AlbumId                                  Title  ArtistId
0        1  For Those About To Rock We Salute You         1
1        2                      Balls to the Wall         2
2        3                      Restless and Wild         2
3        4                      Let There Be Rock         1
4        5                               Big Ones         3
5        6                     Jagged Little Pill         4
6        7                               Facelift         5
7        8                         Warner 25 Anos         6
8        9         Plays Metallica By Four Cellos         7
9       10                             Audioslave         8


### **Question 3: Count the Number of Tracks per Genre**

**Scenario:**  
To plan a marketing campaign, Alex needs to know which genres are most represented in the music store. He asks you to count the number of tracks available for each genre.

**Task:**  
Write a query to count the number of tracks per genre.

---




In [167]:
query = 'SELECT B.Name AS Genre, Count(*) AS Number_of_Tracks FROM Track A \
          INNER JOIN Genre B ON A.GenreId = B.GenreId \
          GROUP BY Genre\
          ORDER BY Number_of_Tracks DESC'

df = pd.read_sql_query(query,conn)
df

Genre  Number_of_Tracks
0                 Rock              1297
1                Latin               579
2                Metal               374
3   Alternative & Punk               332
4                 Jazz               130
5             TV Shows                93
6                Blues                81
7            Classical                74
8                Drama                64
9             R&B/Soul                61
10              Reggae                58
11                 Pop                48
12          Soundtrack                43
13         Alternative                40
14         Hip Hop/Rap                35
15   Electronica/Dance                30
16               World                28
17         Heavy Metal                28
18    Sci Fi & Fantasy                26
19      Easy Listening                24
20              Comedy                17
21          Bossa Nova                15
22     Science Fiction                13
23       Rock And Roll                12
24               Opera                 1

### **Question 4: Calculate the Total Sales by Country**

**Scenario:**  
Alex is interested in understanding the revenue distribution across different countries. He asks you to calculate the total sales generated from each country.

**Task:**  
Write a query to calculate the total sales by country.

In [168]:
query = 'SELECT B.BillingCountry AS Country, SUM(A.UnitPrice) * A.Quantity AS Total_Sales FROM InvoiceLIne A \
        INNER JOIN Invoice B ON A.InvoiceId = B.InvoiceId \
        GROUP BY Country \
        ORDER BY Total_Sales DESC'

df = pd.read_sql_query(query, conn)
df

Country  Total_Sales
0              USA       523.06
1           Canada       303.96
2           France       195.10
3           Brazil       190.10
4          Germany       156.48
5   United Kingdom       112.86
6   Czech Republic        90.24
7         Portugal        77.24
8            India        75.26
9            Chile        46.62
10         Ireland        45.62
11         Hungary        45.62
12         Austria        42.62
13         Finland        41.62
14     Netherlands        40.62
15          Norway        39.62
16          Sweden        38.62
17           Spain        37.62
18          Poland        37.62
19           Italy        37.62
20         Denmark        37.62
21         Belgium        37.62
22       Australia        37.62
23       Argentina        37.62


### **Question 5: Identify the Top 5 Customers by Total Purchase Amount**

**Scenario:**  
To improve customer relationship management, Alex wants to identify the top spenders. He asks you to find out who the top 5 customers are based on their total purchase amount.

**Task:**  
Write a query to identify the top 5 customers by total purchase amount.

---


In [169]:
query = 'SELECT C.CustomerId, C.FirstName, C.LastName, SUM(A.UnitPrice * A.Quantity)  AS Total_Purchase FROM InvoiceLine A \
        INNER JOIN Invoice B ON A.InvoiceId = B.InvoiceId \
        INNER JOIN Customer C ON B.CustomerId = C.CustomerId \
        GROUP BY FirstName, LastName \
        ORDER BY Total_Purchase DESC LIMIT 5'

df = pd.read_sql_query(query,conn)
df

CustomerId FirstName    LastName  Total_Purchase
0           6    Helena        Holý           49.62
1          26   Richard  Cunningham           47.62
2          57      Luis       Rojas           46.62
3          46      Hugh    O'Reilly           45.62
4          45  Ladislav      Kovács           45.62


### **Question 6: Find the Most Popular Artist by Total Track Sales**

**Scenario:**  
Alex wants to identify the most popular artist in the store based on track sales. He asks you to find out which artist has the highest total track sales.

**Task:**  
Write a query to find the most popular artist by total track sales.

---


In [170]:
query = 'SELECT D.ArtistId, D.Name AS Artist_Name, SUM(A.UnitPrice * A.Quantity)  AS Total_Track_Sales From InvoiceLine A \
        INNER JOIN Track B ON A.TrackId = B.TrackId INNER JOIN Album C ON B.AlbumId = C.AlbumId \
        INNER JOIN Artist D ON C.ArtistId = D.ArtistId\
        GROUP BY D.Name \
        ORDER BY Total_Track_Sales DESC LIMIT 5'

df = pd.read_sql_query(query, conn)
df

ArtistId   Artist_Name  Total_Track_Sales
0        90   Iron Maiden             138.60
1       150            U2             105.93
2        50     Metallica              90.09
3        22  Led Zeppelin              86.13
4       149          Lost              81.59


### **Question 7: Determine the Average Invoice Total by Customer Country**

**Scenario:**  
To understand the purchasing behavior of customers from different countries, Alex asks you to determine the average invoice total for each country.

**Task:**  
Write a query to determine the average invoice total by customer country.

---



In [171]:
query = 'SELECT C.Country AS Customer_Country, C.CustomerId, AVG(B.Total) AS Average_Invoice_Total From InvoiceLine A \
        INNER JOIN Invoice B ON A.InvoiceId = B.InvoiceId \
        INNER JOIN Customer C ON B.CustomerId = C.CustomerId\
        GROUP BY Customer_Country\
        ORDER BY Average_Invoice_Total DESC'

df = pd.read_sql_query(query, conn)
df

Customer_Country  CustomerId  Average_Invoice_Total
0           Ireland          46              11.753158
1           Hungary          45              11.753158
2    Czech Republic           6              11.568947
3             Chile          57              10.937368
4           Austria           7              10.647895
5            Norway           4               9.542632
6               USA          23               9.447490
7       Netherlands          48               9.279474
8           Finland          44               9.226842
9           Germany           2               9.161053
10           France          40               9.063684
11         Portugal          34               9.042632
12            India          59               9.017297
13           Sweden          51               8.963684
14           Canada          14               8.848553
15           Brazil          10               8.826842
16   United Kingdom          52               8.805789
17            Spain          50               8.805789
18           Poland          49               8.805789
19            Italy          47               8.805789
20          Denmark           9               8.805789
21          Belgium           8               8.805789
22        Australia          55               8.805789
23        Argentina          56               8.805789

### **Question 8: Analyze Employee Sales Performance**

**Scenario:**  
Alex wants to evaluate the performance of the sales team. He asks you to analyze the sales performance of each employee.

**Task:**  
Write a query to analyze the sales performance of each employee.

---



In [172]:
query = 'SELECT * FROM invoiceline'
query = 'SELECT D.FirstName, D.LastName, C.Country, SUM(A.UnitPrice * A.Quantity)  AS Total_Sales FROM Invoiceline A \
        INNER JOIN Invoice B ON A.InvoiceId = B.InvoiceId\
        INNER JOIN Customer C ON B.CustomerId = C.CustomerId\
        INNER JOIN Employee D ON C.Country = D.Country\
        GROUP BY D.FirstName, D.LastName\
        ORDER BY Total_Sales DESC'

df = pd.read_sql_query(query,conn)
df

FirstName  LastName Country  Total_Sales
0    Andrew     Adams  Canada       303.96
1      Jane   Peacock  Canada       303.96
2     Laura  Callahan  Canada       303.96
3  Margaret      Park  Canada       303.96
4   Michael  Mitchell  Canada       303.96
5     Nancy   Edwards  Canada       303.96
6    Robert      King  Canada       303.96
7     Steve   Johnson  Canada       303.96

### **Question 9: Identify Trends in Monthly Sales**

**Scenario:**  
Alex wants to identify sales trends over time. He asks you to analyze the monthly sales data to spot any trends.

**Task:**  
Write a query to identify trends in monthly sales.

---



In [173]:
query = 'SELECT * FROM Invoice'
query = 'SELECT B.InvoiceDate, SUM(A.Quantity * A.Unitprice) AS Total_Sales \
        FROM InvoiceLine A INNER JOIN Invoice B ON A.InvoiceId = B.InvoiceId\
        GROUP BY B.InvoiceDate\
        ORDER BY Total_Sales DESC'

df = pd.read_sql_query(query,conn)
df.head(20)

InvoiceDate  Total_Sales
0   2013-11-13 00:00:00        25.86
1   2012-08-05 00:00:00        23.86
2   2011-04-28 00:00:00        21.86
3   2010-02-18 00:00:00        21.86
4   2011-05-29 00:00:00        18.86
5   2010-01-18 00:00:00        18.86
6   2010-01-13 00:00:00        17.91
7   2012-10-06 00:00:00        16.86
8   2012-09-05 00:00:00        16.86
9   2011-06-29 00:00:00        15.86
10  2010-03-21 00:00:00        15.86
11  2011-04-23 00:00:00        14.91
12  2013-12-14 00:00:00        13.86
13  2013-10-13 00:00:00        13.86
14  2013-09-12 00:00:00        13.86
15  2013-08-12 00:00:00        13.86
16  2013-07-12 00:00:00        13.86
17  2013-06-11 00:00:00        13.86
18  2013-05-11 00:00:00        13.86
19  2013-04-10 00:00:00        13.86

### **Question 10: Recommend Marketing Strategies Based on Data Insights**

**Scenario:**  
Based on the insights you have gathered from the previous queries, Alex asks you to recommend at least two marketing strategies that MusicWave Inc. should consider to increase sales and customer satisfaction.

**Task:**  
Based on your analysis, write a brief recommendation for two marketing strategies that could help MusicWave Inc. increase sales and improve customer satisfaction.

1. The highest Average Invoice Total (AIT) by Customer Country were Ireland and Hungary with AIT of 11.753158.

2. The Most Popular Artist by Total Track Sales is Iron Maiden with total track sales of 138.60.

3. The country with the highest total sales was USA with total sales of 523.06.

4. Rock is the genre with the highest number of tracks.

5. The customer with the highest Track purchase is Helena Holý.